### DISCUSSION TITLE
Stone liths operation

### DISCUSSION THREAD

**Discussion by Branikolog** [ihhub/fheroes2#5625] [General]
Since we've added the ability for hero to enter the object with the spacebar it's time to ask a question: how should this mechanic operate regarding stone liths. In the OG we cannot come back through stone liths, if there's someone guarding it nearby. While with the current implementatin we can. This completely changes the rules for the majority of maps with stone liths. I think, we should decide, whether we allow player to enter back stone liths without stepping out of it, or forbid this action for stone liths only to force him to move, so he need to fight the guardians. Otherwise, I think we should add penalty at least for entering stone liths, so player won't jump there infinitely. 
Also, I've noticed that AI can't use spacebar to enter object he stands on, and it's really unfair, since he need to fight guardians, while human can avoid battle in front of stone liths. Human and AI heroes should operate exactly the same in these circumstances.

**Comment by LeHerosInconnu** [comment]
Hello @Branikolog,

> Since we've added the ability for hero to enter the object with the spacebar it's time to ask a question: how should this mechanic operate regarding stone liths. In the OG we cannot come back through stone liths, if there's someone guarding it nearby. While with the current implementatin we can. This completely changes the rules for the majority of maps with stone liths. I think, we should decide, whether we allow player to enter back stone liths without stepping out of it, or forbid this action for stone liths only to force him to move, so he need to fight the guardians. Otherwise, I think we should add penalty at least for entering stone liths, so player won't jump there infinitely. Also, I've noticed that AI can't use spacebar to enter object he stands on, and it's really unfair, since he need to fight guardians, while human can avoid battle in front of stone liths. Human and AI heroes should operate exactly the same in these circumstances.

I think that using Stone Liths with the space bar should consume movement points.
This can be done in an elegant way:
When the player uses the space bar, the game automatically counts the movement points needed for the hero to get out of the Stone Liths square and back to the Stone Liths square, thus replacing the player's action of doing it "manually" on the adventure map.
Of course, the penalty of the terrain where the Stone Liths is located as well as the least penalizing terrain penalty for the hero of any of the accessible squares adjacent to the Stone Liths and the hero's orientation skill are also taken into account.

And in the case of the hero who could not return to the square of the Stone Liths (if he had conventionally exited it) because wandering troops are close to the Stone Liths, a message window is displayed when the player uses the space bar to tell him so and ask him if he wishes to attack the wandering creatures.
Something like this:
"Stone Liths.

A group of Goblins is guarding the Stone Liths.
Do you want to attack them?

'Yes' - 'No'"

**Reply by Branikolog** [reply]
Hi, @LeHerosInconnu .
It looks really complicated... So many actions and logic are required to implement this... But the logic still would be not obvious for a player... I'm really concerning about do we really need to make it work in such complex way... Would it be better to leave stone liths operation like 100% the same to the OG?

**Reply by LeHerosInconnu** [reply]
@Branikolog.

> Hi, @LeHerosInconnu . It looks really complicated... So many actions and logic are required to implement this... But the logic still would be not obvious for a player... I'm really concerning about do we really need to make it work in such complex way... Would it be better to leave stone liths operation like 100% the same to the OG?

Personally, I have nothing against it. :)
And the same should be done for the Whirlpool.
